R/ResampleInstances.R

```diff
@@ -8,10 +8,52 @@ instantiateResampleInstance.HoldoutDesc = function(desc, size, task = NULL) {
 }
 
 instantiateResampleInstance.CVDesc = function(desc, size, task = NULL) {
-  if (desc$iters > size)
-    stopf("Cannot use more folds (%i) than size (%i)!", desc$iters, size)
-  test.inds = chunk(seq_len(size), shuffle = TRUE, n.chunks = desc$iters)
-  makeResampleInstanceInternal(desc, size, test.inds = test.inds)
+
+  # Random sampling CV
+  if (!desc$fixed) {
+    if (desc$iters > size) {
+      stopf("Cannot use more folds (%i) than size (%i)!", desc$iters, size)
+    }
+    test.inds = chunk(seq_len(size), shuffle = TRUE, n.chunks = desc$iters)
+    makeResampleInstanceInternal(desc, size, test.inds = test.inds)
+  } else {
+
+    # CV with only predefined indices ("grouping")
+
+    if(is.null(task$blocking)) {
+      stopf("To use blocking in resampling, you need to pass a factor variable when creating the task!")
+    }
+
+    if (desc$iters != length(levels(task$blocking))) {
+      desc$iters = length(levels(task$blocking))
```

Comment:
Why is the number of iterations being adjusted here? There should be a warning.

Reply:
Two reasons: 

1. If the user inserts a different number than levels used for `fixed` in `makeResampleDesc()`, the function will error. 
2. In the inner call, the function is able to adapt by automatically reducing one level. So having always `length(iters) = length(levels(task$blocking)` is the most safe environment for the function to work. 

I added the explanation as a comment.